operator/pkg/clusterconfiguration/cel_patcher_test.go

```diff
@@ -0,0 +1,174 @@
+package clusterconfiguration
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/utils/ptr"
+)
+
+func TestCelPatcher(t *testing.T) {
```

Comment:
Not a request for this PR but if you find yourself revisiting this test suite, it looks like a prime candidate for tests similar to those in the [redpanda helm chart](https://github.com/redpanda-data/redpanda-operator/blob/main/charts/redpanda/testdata/template-cases.txtar) or the slightly less bespoke [datadriven library](https://github.com/cockroachdb/datadriven)

Reply:
Gotcha, I'll do that (as a followup)